pkg/cloudprovider/providers/azure/azure_instances.go

```diff
@@ -29,6 +29,16 @@ import (
 
 // NodeAddresses returns the addresses of the specified instance.
 func (az *Cloud) NodeAddresses(name types.NodeName) ([]v1.NodeAddress, error) {
+	if az.UseInstanceMetadata {
+		text, err := QueryMetadataText("instance/network/interface/0/ipv4/ipAddress/0/privateIpAddress")
```

Comment:
if this errors, should it fall back to the previous API calls or return an error?